Elixir Support

dbernazal opened the pull request.

opcode81 commented:
Thanks for the contribution! 
Please resolve the conflicts (note that `pytest.ini` was removed) and also add Elixir to the list of languages in `README.md`.

dbernazal commented:
Got it, will address.

dbernazal commented:
Still working through the CI errors. I also realized next-ls doesn't support Windows. I could explicitly note this and remove the Elixir tests from the Windows builds if we're ok with that or potentially try to switch it to a different Language server.

opcode81 commented:
> I also realized next-ls doesn't support Windows. 

next-ls does seem to support Windows. The [latest release](https://github.com/elixir-tools/next-ls/releases/tag/v0.23.3) contains a .exe file, at least.

dbernazal commented:
>I added clojure to the list of languages in the readme, there's now a tiny conflict. Pls fix it so we can run the GH actions
Y'all are so fast on the reviews! 😆 

I attempted to re-implement with ElixirLS but it requires Dialyzer to be run in order to have symbol references. This didn't seem great because of the time it takes for Dialyzer to run it's initial pass(can be minutes). 

Instead, I fixed up tests for the NextLS implementation and remove the test run for Windows. I'll monitor the CI and fix anything else that pops up.

opcode81 commented:
> remove the test run for Windows.

@dbernazal but doesn't next-ls support Windows?

MischaPanch commented:
The removing of the tests on windows has to happen with skipif, not just in CI (otherwise tests don't run through locally). Have a look how it was done for closure.

That is, if it can't be made to work for windows. You might want to make the skipif conditional on next-ls being available

MischaPanch commented:
@dbernazal I'm gonna merge this now since I want to move one with #265 , and I don't want to force you to resolve conflicts because of that. If you have time, I'd appreciate a follow-up PR that addresses the comments on tests. Otherwise, I will try addressing them myself soon.

Thank you for bringing Elixir to Serena!

dbernazal commented:
Apologies for the delay, but yes will address the comments in a new PR.
